### ISSUE TITLE
Invalid fix for `PYI053` for f-strings

### ISSUE THREAD

**Issue by dhruvmanila** [astral-sh/ruff#10307]
Given:

```python
def foo(
    arg2: str = f'51 character {x} stringgggggggggggggggggggggggggggggggggggggggggggg',
) -> None: ...
```

Here, the number of characters after the replacement field is > 50 which is what we flag and fix:

```
src/PYI053.pyi:3:35: PYI053 [*] String and bytes literals longer than 50 characters are not permitted
  |
1 | def foo(
2 |     arg1: str = '51 character stringggggggggggggggggggggggggggggggg',
3 |     arg2: str = f'51 character {x} stringgggggggggggggggggggggggggggggggggggggggggggg',
  |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PYI053
4 | ) -> None: ...
  |
  = help: Replace with `...`

ℹ Safe fix
1 1 | def foo(
2 2 |     arg1: str = '51 character stringggggggggggggggggggggggggggggggg',
3   |-    arg2: str = f'51 character {x} stringgggggggggggggggggggggggggggggggggggggggggggg',
  3 |+    arg2: str = f'51 character {x}...',
4 4 | ) -> None: ...

Found 1 error.
[*] 1 fixable with the --fix option.
```

This is not ideal and it raises an interesting question - how to count the characters for an f-string containing replacement fields?
1. `f"one{expr}one"` (6 total or 3 and 3)
2. `"one" f"one{expr}one"` (9 total or 3, 3, 3)
3. `f"one{expr}one" f"one{expr}one"` (12 total or 3, 3, 3, 3)

**Comment by AlexWaygood** [comment]
(I don't think there's a use case for having f-strings in a `.pyi` file, FWIW, so I think the correctness issue with this specific rule is _unlikely_ to come up in real life. The broader issue you're pointing to is obviously still valid, however :)

**Comment by zanieb** [comment]
This is such a weird "safe fix". Does this actually have any effect? Will an IDE hint the wrong default value now?

**Comment by dhruvmanila** [comment]
> Will an IDE hint the wrong default value now?

Yeah, Ruff will basically replace only part of a f-string. The linked PR has a solution to instead count all of the characters in a f-string and replace the _entire_ f-string if it's > 50 instead.
